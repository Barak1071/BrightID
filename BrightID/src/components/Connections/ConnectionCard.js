// @flow

import * as React from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { connect } from 'react-redux';
import moment from 'moment';
import Ionicon from 'react-native-vector-icons/Ionicons';

/**
 * Connection Card in the Connections Screen
 * is created from an array of connections
 * each connection should have:
 * @prop name
 * @prop trustScore
 * @prop connectionTime
 * @prop avatar
 */

type Props = {
  nameornym: string,
  avatar: string,
  trustScore: string,
  connectionDate: string,
  renderActionComponent: () => React.Component,
  publicKey: string,
  style: {},
};

class ConnectionCard extends React.Component<Props> {
  render() {
    const {
      avatar,
      nameornym,
      trustScore,
      connectionDate,
      publicKey,
      style,
    } = this.props;
    const image = avatar
      ? { uri: avatar }
      : require('../../static/default_avatar.jpg');
    return (
      <View style={{ ...styles.container, ...style }}>
        <Image source={image} style={styles.avatar} />
        <View style={styles.info}>
          <Text style={styles.name}>{nameornym}</Text>
          <Text style={styles.trustScore}>{trustScore}% Trusted</Text>
          <Text style={styles.connectedText}>
            Connected {moment(parseInt(connectionDate, 10)).fromNow()}
          </Text>
        </View>
        <TouchableOpacity
          style={styles.moreIcon}
          onPress={this.handleUserOptions}
        >
          <Ionicon size={48} name="ios-more" color="#ccc" />
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    backgroundColor: '#fff',
    height: 94,
    marginBottom: 11.8,
    shadowColor: 'rgba(0,0,0,0.32)',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.43,
    shadowRadius: 4,
  },
  avatar: {
    borderRadius: 30,
    width: 60,
    height: 60,
    marginLeft: 14,
  },
  info: {
    marginLeft: 25,
    flex: 1,
    height: 71,
    flexDirection: 'column',
    justifyContent: 'space-evenly',
  },
  name: {
    fontFamily: 'ApexNew-Book',
    fontSize: 20,
    shadowColor: 'rgba(0,0,0,0.32)',
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 4,
  },
  trustScore: {
    fontFamily: 'ApexNew-Medium',
    fontSize: 14,
    color: 'green',
  },
  connectedText: {
    fontFamily: 'ApexNew-Book',
    fontSize: 14,
  },
  moreIcon: {
    marginRight: 16,
  },
});

export default connect(null)(ConnectionCard);
