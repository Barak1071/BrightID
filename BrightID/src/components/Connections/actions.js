import { AsyncStorage } from 'react-native';
import { defaultSort } from './sortingUtility';
import { setConnections } from '../../actions';
import { objToUint8 } from '../../utils/encoding';

export const getConnections = async () => {
  try {
    const { dispatch } = this.props;
    /**
     * obtain connection keys from async storage
     * currently everything in async storage except for `userData` is a connection
     *
     * THIS MIGHT CHANGE WHEN GROUPS ARE ADDED
     */

    const allKeys = await AsyncStorage.getAllKeys();
    const connectionKeys = allKeys.filter((val) => val !== 'userData');
    const storageValues = await AsyncStorage.multiGet(connectionKeys);
    const connectionValues = storageValues.map((val) => JSON.parse(val[1]));
    const connections = connectionValues.map((val) => {
      val.publicKey = objToUint8(val.publicKey);
      return val;
    });
    // update redux store
    dispatch(setConnections(connections));
    // sort connections
    dispatch(defaultSort());
  } catch (err) {
    console.log(err);
  }
};
