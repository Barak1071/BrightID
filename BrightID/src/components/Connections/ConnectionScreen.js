// @flow

import React, { useCallback, useState } from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
  ScrollView,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { SvgXml } from 'react-native-svg';
import { useActionSheet } from '@expo/react-native-action-sheet';
import verificationSticker from '@/static/verification-sticker.svg';
import { useFocusEffect } from '@react-navigation/native';
import api from '@/api/brightId';
import moment from 'moment';
import { photoDirectory } from '../../utils/filesystem';
import { DEVICE_LARGE } from '../../utils/constants';
import { handleFlagging } from './models/flagConnection';
import { fetchConnectionInfo } from '../../utils/fetchConnectionInfo';
import MutualItems from './MutualItems';
import TrustLevelView from './TrustLevelView';
import { setConnectionLevel } from '../../actions/connections';

type ConnectionScreenProps = {
  route: any,
  navigation: any,
};

function ConnectionScreen(props: ConnectionScreenProps) {
  const { route, navigation } = props;
  const { connectionId } = route.params;
  const { showActionSheetWithOptions } = useActionSheet();
  const dispatch = useDispatch();
  const myId = useSelector((state) => state.user.id);
  const connection: connection = useSelector((state: State) =>
    state.connections.connections.find((conn) => conn.id === connectionId),
  );
  const myConnections = useSelector((state) => state.connections.connections);
  const myGroups = useSelector((state) => state.groups.groups);
  const [mutualGroups, setMutualGroups] = useState<Array<group>>([]);
  const [mutualConnections, setMutualConnections] = useState<Array<connection>>(
    [],
  );
  const [verifications, setVerifications] = useState<Array<string>>([]);

  useFocusEffect(
    useCallback(() => {
      const fetchData = async (connectionId) => {
        console.log(`fetching connection info for ${connectionId}`);
        const connectionData = await fetchConnectionInfo({
          brightId: connectionId,
          myConnections,
          myGroups,
        });
        setMutualGroups(connectionData.mutualGroups);
        setMutualConnections(connectionData.mutualConnections);
        setVerifications(connectionData.verifications);
      };
      if (connection) {
        fetchData(connection.id);
      } else {
        setMutualGroups([]);
        setMutualConnections([]);
        setVerifications([]);
      }
    }, [connection, myConnections, myGroups]),
  );

  if (!connection) {
    // connection not there anymore.
    return <Text>connection vanished.</Text>;
  }

  const imageSource = {
    uri: `file://${photoDirectory()}/${connection.photo.filename}`,
  };

  const brightIdVerified = verifications.includes('BrightID');

  const handleTrustLevelChange = async (level: ConnectionLevel) => {
    console.log(`Setting trust level '${level}' for ${connection.name}`);
    await api.addConnection(myId, connection.id, level, undefined, Date.now());
    dispatch(setConnectionLevel(connection.id, level));
  };

  let flaggingOptions = [
    'Flag as Duplicate',
    'Flag as Fake',
    'Flag as Deceased',
    'Join All Groups',
    'Connect to other fake connections',
    'cancel',
  ];
  if (!__DEV__) {
    // remove debug functionality
    flaggingOptions.splice(3, 2);
  }

  const handleFlagBtnClick = () => {
    showActionSheetWithOptions(
      {
        options: flaggingOptions,
        cancelButtonIndex: flaggingOptions.length - 1,
        title: 'What do you want to do?',
        message: `Flagging ${connection.name} will negatively effect their BrightID score, and this flag might be shown to other users.`,
        showSeparators: true,
        textStyle: {
          color: '#2185D0',
          textAlign: 'center',
          width: '100%',
        },
        titleTextStyle: {
          fontSize: DEVICE_LARGE ? 20 : 17,
        },
        messageTextStyle: {
          fontSize: DEVICE_LARGE ? 15 : 12,
        },
      },
      handleFlagging({
        id: connection.id,
        name: connection.name,
        dispatch,
        secretKey: connection.secretKey,
        callback: () => navigation.goBack(),
      }),
    );
  };

  return (
    <View style={styles.container}>
      <View style={styles.profile}>
        <View style={styles.photoContainer}>
          <TouchableWithoutFeedback
            onPress={() => {
              navigation.navigate('FullScreenPhoto', {
                photo: connection.photo,
                base64: true,
              });
            }}
          >
            <Image
              source={imageSource}
              style={styles.photo}
              resizeMode="cover"
              onError={(e) => {
                console.log(e);
              }}
              accessible={true}
              accessibilityLabel="user photo"
            />
          </TouchableWithoutFeedback>
        </View>
        <View style={styles.nameContainer}>
          <Text style={styles.name}>{connection.name}</Text>
          {brightIdVerified && (
            <SvgXml
              style={styles.verificationSticker}
              width="16"
              height="16"
              xml={verificationSticker}
            />
          )}
          <View style={styles.profileDivider} />
          <View style={styles.badges}>
            {brightIdVerified ? (
              <Text style={styles.verified}>verified</Text>
            ) : (
              <Text style={styles.unverified}>unverified</Text>
            )}
          </View>
        </View>
      </View>
      <View style={styles.connectionInfo}>
        <View style={styles.connectionTimestamp}>
          <Text style={styles.connectionTimestampText}>
            {`Connected ${moment(
              parseInt(connection.createdAt, 10),
            ).fromNow()}`}
          </Text>
        </View>
      </View>
      <View style={styles.mutualInfoContainer}>
        <TrustLevelView
          level={connection.level}
          handleChange={handleTrustLevelChange}
        />
      </View>
      <ScrollView>
        <View style={styles.mutualInfoContainer}>
          <MutualItems
            mutualItems={mutualGroups}
            header="Mutual Groups"
            pluralItemName="groups"
            singularItemName="group"
          />
        </View>
        <View style={styles.mutualInfoContainer}>
          <MutualItems
            mutualItems={mutualConnections}
            header="Mutual Connections"
            pluralItemName="connections"
            singularItemName="connection"
          />
        </View>
        <TouchableOpacity
          testID="FlagBtn"
          style={styles.flagBtn}
          onPress={handleFlagBtnClick}
        >
          <Text style={styles.flagBtnText}>Flag this person</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const ORANGE = '#ED7A5D';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: '8%',
  },
  profile: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: '5%',
    paddingRight: '5%',
  },
  photoContainer: {
    margin: 10,
  },
  photo: {
    borderRadius: 50,
    width: 100,
    height: 100,
  },
  nameContainer: {
    flexDirection: 'column',
    alignItems: 'center',
    marginLeft: 20,
  },
  name: {
    fontFamily: 'Poppins',
    fontWeight: '700',
    fontSize: 17,
    color: '#000',
  },
  profileDivider: {
    borderBottomWidth: 2,
    borderBottomColor: ORANGE,
    paddingBottom: 3,
    width: '118%',
  },
  verificationSticker: {
    marginLeft: DEVICE_LARGE ? 8 : 5,
  },
  badges: {},
  verified: {
    fontFamily: 'Poppins',
    fontWeight: '500',
    color: ORANGE,
    borderWidth: 1,
    borderColor: ORANGE,
    borderRadius: 10,
    marginTop: 6,
    paddingTop: 1,
    paddingBottom: 1,
    paddingLeft: 23,
    paddingRight: 23,
    fontSize: DEVICE_LARGE ? 11 : 10,
  },
  unverified: {
    fontFamily: 'Poppins',
    fontWeight: '500',
    color: '#707070',
    borderWidth: 1,
    borderColor: '#707070',
    borderRadius: 10,
    marginTop: 6,
    paddingTop: 1,
    paddingBottom: 1,
    paddingLeft: 20,
    paddingRight: 20,
    fontSize: DEVICE_LARGE ? 11 : 10,
  },
  connectionInfo: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'center',
  },
  connectionTimestamp: {
    width: '100%',
  },
  connectionTimestampText: {
    fontFamily: 'Poppins',
    fontWeight: '700',
    fontSize: 10,
    color: ORANGE,
  },
  mutualInfoContainer: {
    marginTop: 10,
    marginBottom: 10,
  },
  flagBtn: {
    width: '100%',
    borderRadius: 100,
    borderColor: ORANGE,
    borderWidth: 1,
    backgroundColor: '#fff',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: 12,
    paddingBottom: 12,
  },
  flagBtnText: {
    fontFamily: 'Poppins',
    fontWeight: '700',
    fontSize: DEVICE_LARGE ? 17 : 15,
    color: ORANGE,
    marginLeft: DEVICE_LARGE ? 10 : 8,
  },
});

export default ConnectionScreen;
