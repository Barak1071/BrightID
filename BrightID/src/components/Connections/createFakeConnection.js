// @flow

import { Alert } from 'react-native';
import store from '../../store';
import { addFakeConnection } from '../../actions/fakeContact';

export const createFakeConnection = (navigation: navigation) => () => {
  if (__DEV__) {
    Alert.alert(
      'New Connection',
      'Would you like simulate adding a new connection?',
      [
        {
          text: 'Cancel',
          onPress: () => console.log('Cancel Pressed'),
          style: 'cancel',
        },
        {
          text: 'Sure',
          onPress: () => {
            store.dispatch(addFakeConnection(navigation));
          },
        },
      ],
      { cancelable: true },
    );
  }
};
