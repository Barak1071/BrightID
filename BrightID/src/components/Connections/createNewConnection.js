// @flow

import { Alert } from 'react-native';
import store from '../../store';
import { addConnection } from '../../actions/fakeContact';

export const createNewConnection = (navigation) => () => {
  if(__DEV__) {
    Alert.alert(
      'New Connection',
      'Would you like simulate adding a new connection?',
      [
        {
          text: 'Cancel',
          onPress: () => console.log('Cancel Pressed'),
          style: 'cancel',
        },
        {
          text: 'Sure',
          onPress: () => {
            store.dispatch(addConnection(navigation));
            // navigation.navigate('PreviewConnection');
          },
        },
      ],
      { cancelable: true },
    );
  }
};
