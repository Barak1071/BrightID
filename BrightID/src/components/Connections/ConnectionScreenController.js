// @flow

import React, { useCallback, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { useFocusEffect } from '@react-navigation/native';
import { fetchConnectionInfo } from '../../utils/fetchConnectionInfo';
import ConnectionScreen from './ConnectionScreen';

type ConnectionScreenProps = {
  route: any,
  navigation: any,
};

function ConnectionScreenController(props: ConnectionScreenProps) {
  const { route, navigation } = props;
  const { connectionId } = route.params;
  const connection: connection = useSelector((state: State) =>
    state.connections.connections.find((conn) => conn.id === connectionId),
  );
  const myConnections = useSelector((state) => state.connections.connections);
  const myGroups = useSelector((state) => state.groups.groups);
  const [mutualGroups, setMutualGroups] = useState<Array<group>>([]);
  const [mutualConnections, setMutualConnections] = useState<Array<connection>>(
    [],
  );
  const [verifications, setVerifications] = useState<Array<string>>([]);

  useFocusEffect(
    useCallback(() => {
      const fetchData = async (connectionId) => {
        console.log(`fetching connection info for ${connectionId}`);
        const connectionData = await fetchConnectionInfo({
          brightId: connectionId,
          myConnections,
          myGroups,
        });
        setMutualGroups(connectionData.mutualGroups);
        setMutualConnections(connectionData.mutualConnections);
        setVerifications(connectionData.verifications);
      };
      if (connection) {
        fetchData(connection.id);
      } else {
        setMutualGroups([]);
        setMutualConnections([]);
        setVerifications([]);
      }
    }, [connection, myConnections, myGroups]),
  );

  useEffect(() => {
    if (!connection) {
      // connection not there anymore.
      navigation.goBack();
    }
  }, [navigation, connection]);

  if (!connection) {
    return null;
  }

  const brightIdVerified = verifications.includes('BrightID');

  return (
    <ConnectionScreen
      navigation={navigation}
      brightIdVerified={brightIdVerified}
      connection={connection}
      mutualConnections={mutualConnections}
      mutualGroups={mutualGroups}
    />
  );
}

export default ConnectionScreenController;
