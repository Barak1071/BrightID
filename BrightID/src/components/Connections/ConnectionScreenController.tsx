import React, {
  useCallback,
  useContext,
  useEffect,
  useLayoutEffect,
  useState,
} from 'react';
import { useDispatch, useSelector } from '@/store';
import {
  useFocusEffect,
  useNavigation,
  useRoute,
  RouteProp,
} from '@react-navigation/native';
import ConnectionTestButton from '@/utils/connectionTestButton';
import {
  selectConnectionById,
  selectAllConnections,
  setConnectionVerifications,
} from '@/reducer/connectionsSlice';
import { NodeApiContext } from '@/components/NodeApiGate';
import ConnectionScreen from './ConnectionScreen';

type ConnectionRoute = RouteProp<
  { Connection: { connectionId: string } },
  'Connection'
>;

function ConnectionScreenController() {
  const navigation = useNavigation();
  const route = useRoute<ConnectionRoute>();
  const dispatch = useDispatch();
  const { connectionId } = route.params;
  const api = useContext(NodeApiContext);
  const connection = useSelector((state: State) =>
    selectConnectionById(state, connectionId),
  );
  const myConnections = useSelector(selectAllConnections);
  const myGroups = useSelector((state: State) => state.groups.groups);
  const [mutualGroups, setMutualGroups] = useState<Array<Group>>([]);
  const [mutualConnections, setMutualConnections] = useState<Array<Connection>>(
    [],
  );
  const [connectedAt, setConnectedAt] = useState(0);
  const [loading, setLoading] = useState(false);
  const [connectionProfile, setConnectionProfile] = useState<
    ProfileInfo | undefined
  >(undefined);

  useFocusEffect(
    useCallback(() => {
      const fetchData = async (connectionId) => {
        setLoading(true);
        console.log(`fetching connection info for ${connectionId}`);
        const profile: ProfileInfo = await api.getProfile(connectionId);
        setConnectionProfile(profile);
        setLoading(false);
      };
      if (connectionId !== undefined) {
        fetchData(connectionId);
      }
    }, [api, connectionId]),
  );

  // Update connection verifications in store
  useEffect(() => {
    if (connectionProfile) {
      console.log(`Updating verifications for ${connectionProfile.id}`);
      dispatch(
        setConnectionVerifications({
          id: connectionProfile.id,
          verifications: connectionProfile.verifications,
        }),
      );
    }
  }, [connectionProfile, dispatch]);

  // Update mutual groups etc. in local state
  useEffect(() => {
    if (connectionProfile) {
      console.log(`Updating mutual groups etc. for ${connectionProfile.id}`);
      setConnectedAt(connectionProfile.connectedAt);
      setMutualConnections(
        myConnections.filter((conn) => {
          return connectionProfile.mutualConnections.includes(conn.id);
        }),
      );
      setMutualGroups(
        myGroups.filter((g) => {
          return connectionProfile.mutualGroups.includes(g.id);
        }),
      );
    }
  }, [connectionProfile, myConnections, myGroups]);

  useEffect(() => {
    if (!connection) {
      // connection not there anymore.
      navigation.goBack();
    }
  }, [navigation, connection]);

  // Add fake user functionality in DEV mode
  useLayoutEffect(() => {
    if (__DEV__) {
      navigation.setOptions({
        headerRight: () => <ConnectionTestButton connectionId={connectionId} />,
      });
    }
  }, [navigation, connectionId]);

  if (!connection) {
    return null;
  }

  const brightIdVerified = connection.verifications?.some(
    (v) => v?.name === 'BrightID',
  );
  const verifiedAppsCount = connection.verifications?.filter((v) => {
    if ('app' in v) {
      return v.app;
    } else return false;
  }).length;

  return (
    <ConnectionScreen
      connection={connection}
      brightIdVerified={brightIdVerified}
      verifiedAppsCount={verifiedAppsCount}
      loading={loading}
      connectedAt={connectedAt}
      mutualConnections={mutualConnections}
      mutualGroups={mutualGroups}
    />
  );
}

export default ConnectionScreenController;
