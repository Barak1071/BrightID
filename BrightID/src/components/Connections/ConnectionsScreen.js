// @flow

import * as React from 'react';
import { AsyncStorage, StyleSheet, View, Alert } from 'react-native';
import { connect } from 'react-redux';
import HeaderButtons, {
  HeaderButton,
  Item,
} from 'react-navigation-header-buttons';
import Material from 'react-native-vector-icons/MaterialCommunityIcons';
import SearchConnections from './SearchConnections';
import ConnectionCard from './ConnectionCard';
import { NavigationEvents } from 'react-navigation';
import { getConnections } from '../../actions/getConnections';
import { createNewConnection } from './createNewConnection';
import emitter from '../../emitter';
import BottomNav from '../BottomNav';
import { renderListOrSpinner } from './renderConnections';
import api from '../../Api/brightId';

/**
 * Connection screen of BrightID
 * Displays a search input and list of Connection Cards
 */

// header Button
const MaterialHeaderButton = (passMeFurther) => (
  <HeaderButton
    {...passMeFurther}
    IconComponent={Material}
    iconSize={32}
    color="#fff"
  />
);

type Props = {
  connections: Array<{
    nameornym: string,
    id: number,
  }>,
  searchParam: string,
  dispatch: (() => Promise<null>) => Promise<null>,
};

type State = {
  loading: boolean,
};

class ConnectionsScreen extends React.Component<Props, State> {
  static navigationOptions = ({ navigation }) => ({
    title: 'Connections',
    headerRight: (
      <HeaderButtons HeaderButtonComponent={MaterialHeaderButton}>
        <Item
          title="options"
          iconName="dots-horizontal"
          onPress={createNewConnection(navigation)}
        />
      </HeaderButtons>
    ),
  });

  state = {
    loading: true,
  };

  componentDidMount() {
    this.getConnections();
    emitter.on('refreshConnections', this.getConnections);
    emitter.on('removeConnection', this.removeConnection);
  }

  componentWillUnmount() {
    emitter.off('refreshConnections', this.getConnections);
    emitter.off('removeConnection', this.removeConnection);
  }

  getConnections = async () => {
    const { dispatch } = this.props;
    await dispatch(getConnections());
    this.setState({
      loading: false,
    });
  };

  removeConnection = async (publicKey) => {
    try {
      await api.deleteConnection(publicKey);
      // remove connection from async storage
      await AsyncStorage.removeItem(JSON.stringify(publicKey));
      emitter.emit('refreshConnections', {});
    } catch (err) {
      Alert("Couldn't remove connection", err.stack);
    }
  };

  filterConnections = () => {
    const { connections, searchParam } = this.props;
    return connections.filter((item) =>
      `${item.nameornym}`
        .toLowerCase()
        .replace(/\s/g, '')
        .includes(searchParam.toLowerCase().replace(/\s/g, '')),
    );
  };

  renderConnection = ({ item }) => <ConnectionCard {...item} />;

  render() {
    const { navigation } = this.props;
    return (
      <View style={styles.container}>
        <NavigationEvents
          onDidFocus={() => {
            this.getConnections();
          }}
        />
        <View style={styles.mainContainer}>
          <SearchConnections navigation={navigation} />
          <View style={styles.mainContainer}>{renderListOrSpinner(this)}</View>
        </View>
        <BottomNav navigation={navigation} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fdfdfd',
  },
  mainContainer: {
    flex: 1,
    backgroundColor: '#fdfdfd',
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 8,
  },
  connectionsContainer: {
    flex: 1,
  },
  moreIcon: {
    marginRight: 16,
  },
  emptyText: {
    fontFamily: 'ApexNew-Book',
    fontSize: 20,
  },
});

export default connect((state) => state.main)(ConnectionsScreen);
