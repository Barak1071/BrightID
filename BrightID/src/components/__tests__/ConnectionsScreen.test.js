import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { ConnectionsScreen } from '../Connections/ConnectionsScreen';

// you can mock any middlewares here if necessary

const props = {
  dispatch: jest.fn(),
  navigation: { navigate: jest.fn() },
  searchParam: '',
  connections: [],
};

describe('Testing ConnectionsScreen', () => {
  it('renders blank connection screen', () => {
    const wrapper = renderer.create(<ConnectionsScreen {...props} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
