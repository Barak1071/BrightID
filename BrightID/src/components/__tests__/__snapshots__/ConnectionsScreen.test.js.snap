// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConnectionsScreen renders blank connection screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fdfdfd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fdfdfd",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 8,
        }
      }
    >
      <SearchConnections
        navigation={
          Object {
            "navigate": [MockFunction],
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fdfdfd",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 8,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "fontFamily": "ApexNew-Book",
                "fontSize": 20,
              }
            }
          >
            No connections
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 25,
          "justifyContent": "center",
          "position": "absolute",
          "right": 25,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "zIndex": 100,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f98961",
              "borderRadius": 27,
              "elevation": 4,
              "height": 54,
              "justifyContent": "center",
              "shadowColor": "rgba(0,0,0,0.5)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowRadius": 4,
              "width": 54,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 36,
                },
                Object {
                  "height": 36,
                  "width": 36,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopColor": "rgba(0, 0, 0, .3)",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "height": 63,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="Home"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Connections"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Connections
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Groups"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Groups
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Apps"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Notifications
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="Apps"
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#222",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "ApexNew-Book",
              "fontSize": 12,
              "marginBottom": 1.5,
            }
          }
        >
          Apps
        </Text>
      </View>
    </View>
  </View>
</View>
`;
